vdr/vdr.go

```diff
@@ -132,32 +132,98 @@ func (r VDR) Resolve(id did.DID, metadata *types.ResolveMetadata) (*did.Document
 func (r VDR) Update(id did.DID, current hash.SHA256Hash, next did.Document, _ *types.DocumentMetadata) error {
 	logging.Log().Debugf("Updating DID Document: %s", id)
 	// TODO: check the integrity / validity of the proposed DID Document.
-	resolverMetada := &types.ResolveMetadata{
+	resolverMetadata := &types.ResolveMetadata{
 		Hash:             &current,
-		AllowDeactivated: false,
+		AllowDeactivated: true,
 	}
-	currentDIDdocument, meta, err := r.store.Resolve(id, resolverMetada)
+	currentDIDDocument, meta, err := r.store.Resolve(id, resolverMetadata)
 	if err != nil {
 		return err
 	}
+	if isDeactivated(currentDIDDocument) {
```

Comment:
I'm a but puzzled why you chose to add additional functionality to the existing PR? This can easily go in a separate PR (even more since the discussion has just started)